recipes/rmlui/4.x/conanfile.py

```diff
@@ -85,78 +86,96 @@ def requirements(self):
+            self.requires("robin-hood-hashing/3.11.3", headers=True, transitive_headers=True)
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
-
-    def _configure_cmake(self):
-        if not hasattr(self, "_cmake"):
-            self._cmake = CMake(self)
-            self._cmake.definitions["BUILD_LUA_BINDINGS"] = self.options.with_lua_bindings
-            self._cmake.definitions["BUILD_SAMPLES"] = False
-            self._cmake.definitions["CUSTOM_CONFIGURATION"] = True
-            self._cmake.definitions["CUSTOM_INCLUDE_DIRS"] = ";".join(self.deps_cpp_info["robin-hood-hashing"].include_paths)
-            self._cmake.definitions["DISABLE_RTTI_AND_EXCEPTIONS"] = not self.options.enable_rtti_and_exceptions
-            self._cmake.definitions["ENABLE_PRECOMPILED_HEADERS"] = True
-            self._cmake.definitions["ENABLE_TRACY_PROFILING"] = False
-            self._cmake.definitions["MATRIX_ROW_MAJOR"] = self.options.matrix_mode == "row_major"
-            self._cmake.definitions["NO_FONT_INTERFACE_DEFAULT"] = self.options.font_interface is None
-            self._cmake.definitions["NO_THIRDPARTY_CONTAINERS"] = not self.options.with_thirdparty_containers
-
-            self._cmake.configure()
-
-        return self._cmake
+        get(self, **self.conan_data["sources"][self.version],
+                  destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.cache_variables["BUILD_LUA_BINDINGS"] = self.options.with_lua_bindings
+        tc.cache_variables["BUILD_SAMPLES"] = False
+        tc.cache_variables["DISABLE_RTTI_AND_EXCEPTIONS"] = not self.options.enable_rtti_and_exceptions
+        tc.cache_variables["ENABLE_PRECOMPILED_HEADERS"] = True
+        tc.cache_variables["ENABLE_TRACY_PROFILING"] = False
+        tc.cache_variables["MATRIX_ROW_MAJOR"] = self.options.matrix_mode == "row_major"
+        tc.cache_variables["NO_FONT_INTERFACE_DEFAULT"] = not self.options.font_interface
+        tc.cache_variables["NO_THIRDPARTY_CONTAINERS"] = not self.options.with_thirdparty_containers
+        tc.generate()
+
+        deps = CMakeDeps(self)
+        deps.generate()
 
     def _patch_sources(self):
-        # The *.cmake files that conan generates using cmake_find_package for CMake's find_package to consume use
-        # different variable naming than described in CMake's documentation, thus the need for most of the replacements.
-        # References:
-        #  * https://cmake.org/cmake/help/latest/module/FindFreetype.html
-        #  * https://cmake.org/cmake/help/latest/module/FindLua.html
+
+        # Since we have switched to CMakeDeps, the targets are named differently, so we need to patch
```

Comment:
With CMakeDeps you can set_property in the generate method to change the names instead of patching them